src/app/components/task-button/task-button.component.ts

```diff
@@ -62,11 +64,30 @@ export class TaskButtonComponent implements OnInit {
       { display: 'inline-block' }
     );
   }
+
   onTaskButtonClick({ dataJob, id }: TaskEventData): void {
+
     switch (dataJob) {
+      case this.constantsService.COMPLETE:    
+        this.taskButtonClick.emit({id, dataJob});
+        break;
+  
       case this.constantsService.DELETE_TODO:
-        this.taskButtonClick.emit({ id, dataJob });
+        this.taskButtonClick.emit({id, dataJob});
         break;
     }
   }
+
+  visibilityHandle({dataJob}: {dataJob: any}, task: Task): boolean {
```

Comment:
Similar method seems used in `add-card-component`. I would reuse them using a service.

Reply:
in add-card-component this function is not needed so I removed it from  add-card component.